### PR TITLE
feat: dry out the header code and implement `Content-Security-Policy`

### PR DESCRIPTION
**Pull request by reidrankin**
## Description

Dry out the header infrastructure and implement `Content-Security-Policy`.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,7 +73,7 @@
     "show-linked-packages": "find node_modules node_modules/@* -depth 1 -type l -print",
     "link-packages": "./scripts/link-packages.sh",
     "unlink-packages": "find node_modules node_modules/@* -depth 1 -type l -print | awk -F/ '{ printf \"%s/%s\\n\",$2,$3; }' | xargs yarn unlink && yarn install --force",
-    "build": "GENERATE_SOURCEMAP=false DISABLE_ESLINT_PLUGIN=true react-app-rewired build",
+    "build": "GENERATE_SOURCEMAP=false DISABLE_ESLINT_PLUGIN=true react-app-rewired build && node headers.js",
```
Comment (cjthompson) [inline package.json:76]: can't you put this in the webpack config, to output a headers file?

Comment (reidrankin) [reply package.json:76]: possibly, but i didn't want to put something non-idempotent in there

**Comment by reidrankin** [comment]
So vercel's basically just not gonna work here. The headers include values from .env, which means they have to be generated, but `vercel.json` just can't be generated. I've tried several hacks to trick it with no success; their runner loads it as the first operation and never re-reads the file.

**Review by cjthompson — APPROVED** [review]

**Review by cjthompson — APPROVED** [review]

File: react-app-rewired.config.js [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
     return (proxy, allowedHost) => {
       const config = configFunction(proxy, allowedHost)
-      config.headers = {
-        'Cross-Origin-Opener-Policy': 'same-origin',
-        'Permissions-Policy': 'document-domain=()',
-        'Referrer-Policy': 'no-referrer',
-        'X-Content-Type-Options': 'nosniff',
-        'X-Frame-Options': 'DENY'
-      }
+      config.headers = require('./headers')
```
Comment (reidrankin) [inline react-app-rewired.config.js:10]:
```suggestion
      config.headers = headers
```

File: react-app-rewired.config.js [hunk 3]
```diff
@@ -1,17 +1,13 @@
 /**
  * React App Rewired Config
  */
+process.env.REACT_APP_CSP = require('./headers')['Content-Security-Policy'] ?? ''
```
Comment (reidrankin) [inline react-app-rewired.config.js:4]:
```suggestion
const headers = require('./headers')
process.env.REACT_APP_CSP = headers['Content-Security-Policy'] ?? ''
```

**Pull request merged**